// webhookit-process/lib/webhookit-process.js

var http = require('http'),
	 querystring = require('querystring'),
    URL = require('url'),
	 xml2js = require('xml2js-expat');


exports.definition = {
    "name": "process",
    "category": "Processes",
    "container": {
        "icon": "/images/icons/wrench.png",
        "xtype": "WireIt.FormContainer",
        "title": "process_step",
        "fields": [
        {
            "type": "string",
            "name": "name",
            "wirable": true,
            label: "Name"
        },
        {
            "type": "list",
            "name": "materials",
            listLabel: 'Materials',
            elementType: {
                type: 'combine',
                fields: [{typeInvite: 'material' , wirable: true}],
                separators: [false]
            }
        }
        ],
        "terminals": [
        {
            "name": "out",
            "direction": [0, 1],
            "offsetPosition": {
                "left": 86,
                "bottom": -15
            },
            "ddConfig": {
                "type": "output",
                "allowedTypes": ["input"]
            }
        }
        ]
    }
};

exports.run = function(params, cb) {
	console.log(params);
	var r = params;

	cb( {out: r} );
};
